userspace/libsinsp/plugin.h

```diff
@@ -166,6 +172,10 @@ class sinsp_plugin
 	std::string get_init_schema(ss_plugin_schema_type& schema_type) const;
 	bool set_config(const std::string& config);
 	std::vector<metrics_v2> get_metrics() const;
+	void capture_open();
+	void capture_close();
+	thread_pool::routine_id_t subscribe_routine(ss_plugin_routine_fn_t routine_fn, ss_plugin_routine_state_t* routine_state);
+	void unsubscribe_routine(thread_pool::routine_id_t routine_id);
```

Comment:
These two should be private methods IMO. Clients of the plugin wrapper should not invoke these at will.